src/object.d

```diff
@@ -3241,6 +3241,269 @@ template RTInfo(T)
     enum RTInfo = null;
 }
 
+private template Floating1(T)
+if (is(T == cfloat) || is(T == cdouble) || is(T == creal))
+{
+    // Use rt.cmath2._Ccmp instead ?
+    int compare(T f1, T f2)
+    {
+        int result;
+
+        if (f1.re < f2.re)
+            result = -1;
+        else if (f1.re > f2.re)
+            result = 1;
+        else if (f1.im < f2.im)
+            result = -1;
+        else if (f1.im > f2.im)
+            result = 1;
+        else
+            result = 0;
+        return result;
+    }
+}
+
+private template Floating1(T)
+if (is(T == float) || is(T == double) || is(T == real))
+{
+    int compare(T d1, T d2)
+    {
+        if (d1 != d1 || d2 != d2) // if either is NaN
+        {
+            if (d1 != d1)
+            {
+                if (d2 != d2)
+                    return 0;
+                return -1;
+            }
+            return 1;
+        }
+        return d1 < d2 ? -1 : (d1 > d2);
+    }
+}
+
+private template Array1(T)
```

Comment:
This, too, should be nestable after the above. There's no need for a template, just write a static function inside _adCmpT.